cli/orchestration/runtime-configuration.md

```diff
@@ -5,46 +5,164 @@ description: Learn how to configure environment variables available to the terra
 
 # Runtime Configuration
 
-The `terramate.config.run.env` block can be used to define a map of environment variables
-that will be set when running a command using `terramate run`.
+Terramate CLI offers flexible options for configuring environment variables for orchestrating commands and managing your infrastructure as code. This guide explains how to set environment variables project-wide, for specific stacks or directories, through the CLI, and using `.direnv`.
 
-The following example exports the `TF_PLUGIN_CACHE_DIR` environment variable to
-enable [Terraform Provider Plugin Caching](https://www.terraform.io/cli/config/config-file#provider-plugin-cache).
+## Configuring Environment Variables
+
+### Using the Current Environment
+You can access any environment variable that is exposed by the host when orchestrating commands with the `terramate run` and `terramate script run` commands.
+
+```sh
+export ENVIRONMENT=prod
+terramate run --eval -- echo "\${terramate.root.path.fs.absolute}/config/${ENVIRONMENT}.tfvars"
+```
+### Passing Variables Directly to the Terramate Process
+You can also pass variables directly to the Terramate process.
+
+```sh
+ENVIRONMENT=prod terramate run --eval -- echo "\${terramate.root.path.fs.absolute}/config/${ENVIRONMENT}.tfvars"
+```
+
+## Project-wide Environment Variables
+To configure project-wide environment variables, define them in the `terramate.config.run.env` block inside the `terramate.tm.hcl` file at the root of your repository.
 
 ```hcl
+# terramate.tm.hcl
+
 terramate {
   config {
     run {
       env {
         TF_PLUGIN_CACHE_DIR = "/some/path/etc"
+        ALICE = "BOB"
       }
     }
   }
 }
 ```
+## Stack-level Environment Variables
+To configure stack-specific environment variables, define them in the `terramate.config.run.env` block inside the `stack.tm.hcl` file within the desired stack.
```

Comment:
So this is sligthly missleading. You can define the block in the `stack.tm.hcl` but technically you can define it anywhere in the project hierarchy, such as directories or stacks + it will be inherited.  I feel we should make this more clear here